core/capabilities/compute/cache.go

```diff
@@ -0,0 +1,123 @@
+package compute
+
+import (
+	"sync"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/wasm/host"
+)
+
+var (
+	moduleCacheHit = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "compute_module_cache_hit",
+		Help: "hit vs non-hits of the module cache for custom compute",
+	}, []string{"hit"})
+	moduleCacheEviction = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "compute_module_cache_eviction",
+		Help: "evictions from the module cache",
+	})
+	moduleCacheAddition = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "compute_module_cache_addition",
+		Help: "additions to the module cache",
+	})
+)
+
+type moduleCache struct {
```

Comment:
I'm lost on how this cache keeps warm instances, and how it ever gets hits based on use. Init module is only called in Execetue, which is only done on a miss.  There's never a pre-warm.

Reply:
Yeah this cache doesn't pre-warm (premature optimization and all that); all it does is keep modules around for a certain amount of time to amortize the initialization